massa-versioning-worker/src/versioning.rs

```diff
@@ -581,24 +618,105 @@ impl MipStoreRaw {
             let added = to_add.keys().cloned().collect();
             let updated = to_update.keys().cloned().collect();
 
-            self.0.append(&mut to_update);
-            self.0.append(&mut to_add);
+            self.store.append(&mut to_update);
+            self.store.append(&mut to_add);
             Ok((updated, added))
         } else {
             Err(())
         }
     }
+
+    fn update_network_version_stats(
+        &mut self,
+        slot_timestamp: MassaTime,
+        network_versions: Option<(u32, u32)>,
+    ) {
+        if let Some((_current_network_version, announced_network_version)) = network_versions {
+            let removed_version_ = match self.stats.latest_announcements.len() {
+                n if n > self.stats.config.block_count_considered => {
+                    self.stats.latest_announcements.pop_front()
```

Comment:
Why have you decided not to use CircularBuffers?

Reply:
I've tried but then I cannot use #derive(Clone, PartialEq) and that was problematic in many places :-/

Reply:
Aw to bad!